### ISSUE TITLE
When no websockets and multi-editors switch user rights to read-only

### ISSUE THREAD

**Issue by virgile-dev** [suitenumerique/docs#486]
## Feature Request

**Is your feature request related to a problem or unsupported use case? Please describe.**
Here is the related issue : https://github.com/numerique-gouv/impress/issues/456

**Describe the solution you'd like**
When a user has websockets blocked AND there are multiple possible editors (the document is shared with other people or there an edit link that allow collaboration) they should be able to only read the document and an alert banner should display to explain why.

**Describe alternatives you've considered**
This is until we implement some kind of alternative solution like polling. (https://github.com/numerique-gouv/impress/issues/437)

**Discovery, Documentation, Adoption, Migration Strategy**
![CleanShot 2024-12-09 at 18 26 13@2x](https://github.com/user-attachments/assets/dbba0cb3-3fee-4e46-8830-dc63127ef8a8)
(full mockup in the "In progress" area.

**Comment by virgile-dev** [comment]
For the banner message Julien suggested a text that I think is really great :
"Votre accès internet ou politique SSI ne permet pas d'utiliser l'édition collaborative, vous pouvez contacter votre DSI pour demander une lever ce blocage"

**Comment by virgile-dev** [comment]
Linked with https://github.com/suitenumerique/docs/issues/456

**Comment by rl-83** [comment]
Here is [my updated proposal](https://www.figma.com/design/qdCWR4tTUr7vQSecEjCyqO/DOCS?node-id=7667-23920&t=TKlbHTcAyL1GBp7c-1).  
The idea is to completely block editing functionality for users who do not have WebSockets.  
Clicking on "learn more" opens a modal that explains in simple terms why it isn't working. In this explanatory text, I suggest mentioning: (1) that the user typically has editing rights on this specific document, but (2) that their network configuration does not allow collaboration, and therefore editing.

**Comment by virgile-dev** [comment]
"La configuration réseau de votre poste de travail ou de votre accès internet de ne permet pas d'utiliser l'édition collaborative. 
Docs utilise les websockets la co-édition en temps réel. Les websockets est un canal qui permet à un site web et à un serveur d'échanger des messages instantanément et de façon bidirectionnel.  Vous pouvez contacter à vos responsables informatiques pour leur demander d'autoriser les websockets."

**Comment by virgile-dev** [comment]
This is the text for the modal:
"The network configuration of your workstation or internet access does not allow collaborative editing. Docs uses WebSockets for real-time co-editing. WebSockets are a channel that enables a website and a server to exchange messages instantly and bidirectionally. You can contact your IT administrators to request that WebSockets be allowed."

"La configuration réseau de votre poste de travail ou de votre accès internet de ne permet pas d'utiliser l'édition collaborative.
Docs utilise les websockets la co-édition en temps réel. Les websockets est un canal qui permet à un site web et à un serveur d'échanger des messages instantanément et de façon bidirectionnel. Vous pouvez contacter à vos responsables informatiques pour leur demander d'autoriser les websockets."

**Comment by virgile-dev** [comment]
@lunika you might want to have a look (according to sam) before we start the dev on this

**Comment by AntoLC** [comment]
To summarize, user will switch to "Reader" mode if :
- The user is online (has internet)
- The user is not connected to the collaboration server
- The doc has members or is "public editable" or "auth editable"

**Comment by sampaccoud** [comment]
> To summarize, user will switch to "Reader" mode if :
> 
> * The user is online (has internet)
> * The user is not connected to the collaboration server
> * The doc has members or is "public editable" or "auth editable"

Could we add the condition that at least one other user is currently connected? This would minimize occurrences of locking the document for users with no websocket.

**Comment by AntoLC** [comment]
> Could we add the condition that at least one other user is currently connected? This would minimize occurrences of locking the document for users with no websocket.

Yes we can know the **connected** users on the server, but we have a case where multiple users **not connected** on the server are editing at the same time to tackle as well so.

**Comment by jidhub** [comment]
"a mechanism to add and remove their presence manually to the collab server (http endpoint)": this seems a wise solution.
